### ISSUE TITLE
FingerprintUSBHost is not compiling

### ISSUE THREAD

**Issue by fletchjp** [keyboardio/FingerprintUSBHost#9]
Hi. I am compiling FingerprintUSBHost on Windows 11 to load to an Arduino UNO. I am getting compiler errors. I am using Arduino IDE 2.3.4 and the latest board 1.8.6

c:\Users\john\Documents\Arduino\libraries\FingerprintUSBHost\src/FingerprintUSBHost.h:38:55: error: expected class-name before '{' token
 class FingerprintUSBHost_ : public PluggableUSBModule {

c:\Users\john\Documents\Arduino\libraries\FingerprintUSBHost\src/FingerprintUSBHost.h:52:23: error: 'USBSetup' has not been declared
     int getDescriptor(USBSetup& setup);

I think that the PluggableUSB has changed as there is no constructor for PluggableUSBModule without arguments.

A number of other people have reported a problem with USBSetup.

My interest is to be able to distinguish between compiles on Windows 11 and Raspberry Pi OS (Debian) as some of my Arduino NANOs use a USB chip which is not supported on Windows 11 and is supported by R Pi OS.

Have you run this code lately?

Thanks

John

**Comment by obra** [comment]
The 328 on an Uno doesn't speak enough USB. You need at least a leonardo.

**Comment by fletchjp** [comment]
Thank you. I have tried several different things, NANO, NANO EVERY, MEGA, DUE. None of them compile. Leonardo does, which I only have as a Raspberry Pi daughter board (DFROBOT DFR0327). Is there anything you can suggest so that I can identify what is uploading to a NANO?

Thanks John

**Comment by obra** [comment]
Not really. The technique requires a 32U4 or other device with a similarly
capable USB stack.  And even then, it's not *reliable*.

**Comment by fletchjp** [comment]
Thank you. I have acquired a Pro Mini and that does work and detects Windows when connected to Windows 11.
